frontend/src/metabase/common/components/FilterPicker/CoordinateFilterPicker/utils.ts

```diff
@@ -1,103 +1,85 @@
 import * as Lib from "metabase-lib";
-import type { CoordinateFilterOperatorName } from "metabase-lib";
 import { OPERATOR_OPTIONS } from "./constants";
+import type { NumberValue } from "./types";
 
-export function findLatitudeColumns(query: Lib.Query, stageIndex: number) {
-  const filterableColumns = Lib.filterableColumns(query, stageIndex);
-  return filterableColumns.filter(column => Lib.isLatitude(column));
+function isNotEmpty(value: NumberValue) {
+  return value !== "";
 }
 
-export function findLongitudeColumns(query: Lib.Query, stageIndex: number) {
-  const filterableColumns = Lib.filterableColumns(query, stageIndex);
-  return filterableColumns.filter(column => Lib.isLongitude(column));
-}
-
-/**
- * For "inside" filters, we may start with just one column, and need to find a second column
- */
-export const findSecondColumn = ({
-  query,
-  stageIndex,
-  column,
-  filter,
-  operatorName,
-}: {
-  query: Lib.Query;
-  stageIndex: number;
-  column: Lib.ColumnMetadata;
-  filter?: Lib.FilterClause;
-  operatorName: CoordinateFilterOperatorName;
-}): Lib.ColumnMetadata | null => {
-  if (operatorName !== "inside") {
-    return null;
-  }
-
-  if (filter) {
-    const filterParts = Lib.coordinateFilterParts(query, stageIndex, filter);
-    if (filterParts?.longitudeColumn) {
-      return filterParts.longitudeColumn;
-    }
+export function getDefaultValues(
+  operator: Lib.CoordinateFilterOperatorName,
+  values: NumberValue[] = [],
+): NumberValue[] {
+  const { valueCount, hasMultipleValues } = OPERATOR_OPTIONS[operator];
+  if (hasMultipleValues) {
+    return values.filter(isNotEmpty);
   }
 
-  if (Lib.isLatitude(column)) {
-    return findLongitudeColumns(query, stageIndex)[0] ?? null;
-  }
+  return Array(valueCount)
+    .fill("")
+    .map((value, index) => values[index] ?? value);
+}
 
-  if (Lib.isLongitude(column)) {
-    return findLatitudeColumns(query, stageIndex)[0] ?? null;
+export function hasValidValues(
+  operator: Lib.CoordinateFilterOperatorName,
+  values: NumberValue[] = [],
+): values is number[] {
+  const { valueCount, hasMultipleValues } = OPERATOR_OPTIONS[operator];
+  if (!values.every(isNotEmpty)) {
```

Comment:
It'd be easier to read without the 2 negations:

```suggestion
  if (values.some(isEmpty)) {
```

_This comment applies to other places with `isNotEmpty` function in this PR._

Reply:
In this case I would need to have both `isEmpty` and `isNotEmpty`, so I'd rather keep it as it is